File: lib/layer2_core/light/e_surface_container_light.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/light/system.dart';
import '../../layer1_raw/light/standard.dart';
import '../../layer1_raw/light/blue.dart';
import '../../layer1_raw/light/green.dart';
import '../../layer1_raw/light/red.dart';
import '../../layer1_raw/light/nothing.dart';

class ESurfaceContainerLight {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyl15;
      case 'standard': return cstl15;
      case 'blue': return cbl15;
      case 'green': return cgl15;
      case 'red': return crl15;
      case 'nothing': return cnol15;
      default: return cstl15;
    }
  }
}

File: android/src/main/kotlin/com/example/design_engine/layer2_core/light/ESurfaceContainerLight.kt
package com.example.design_engine.layer2_core.light
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object ESurfaceContainerLight {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyl15
            AppTheme.STANDARD -> R.color.cstl15
            AppTheme.BLUE -> R.color.cbl15
            AppTheme.GREEN -> R.color.cgl15
            AppTheme.RED -> R.color.crl15
            AppTheme.NOTHING -> R.color.cnol15
        }
    }
}
